command/sub8_mission_control/sub8_mission_control/goal_client.py

```diff
@@ -0,0 +1,153 @@
+        ''' Move backwards in the given amount of time '''
+        to_send = Waypoint()
+        to_send.pose.position.x = self.current_pos.pose.position.x - move
+        # send goal to server to complete within timeout
+        move_result = self.send_goal(to_send, timeout)
+        return move_result
+
+    def move_left(self, move, timeout = 10):
+        ''' Move left in the given amount of time '''
+        to_send = Waypoint()
+        to_send.pose.position.y = self.current_pos.pose.position.y - move
+        # send goal to server to complete within timeout
+        move_result = self.send_goal(to_send, timeout)
+        return move_result
+
+    def move_right(self, move, timeout = 10):
+        ''' Move right in the given amount of time '''
+        to_send = Waypoint()
+        to_send.pose.position.y = self.current_pos.pose.position.y + move
+        # send goal to server to complete within timeout
+        move_result = self.send_goal(to_send, timeout)
+        return move_result
+
+    def move_up(self, move, timeout = 10):
+        ''' Move up in the given amount of time '''
+        to_send = Waypoint()
+        to_send.pose.position.z = self.current_pos.pose.position.z + move
+        # send goal to server to complete within timeout
+        move_result = self.send_goal(to_send, timeout)
+        return move_result
+
+    def move_down(self, move, timeout = 10):
+        ''' Move down in the given amount of time '''
+        to_send = Waypoint()
+        to_send.pose.position.z = self.current_pos.pose.position.z - move
+        # send goal to server to complete within timeout
+        move_result = self.send_goal(to_send, timeout)
+        return move_result
+
+    def absolute_move(self, x=None, y=None, z=None, qx=None, qy=None, qz=None, qw=None, timeout=10):
+        ''' 
+            Move to an absolute /map position 
+            The user should specify position with
+            class_instance.absolute_goal and then call this function
+        '''
+
+        if x == None: x = self.current_pos.pose.x 
+        if y == None: y = self.current_pos.pose.y,
+        if z == None: z = self.current_pos.pose.z, 
+        if qx == None: qx = self.current_pos.pose.orientation.x,
+        if qy == None: qy = self.current_pos.pose.orientation.y, 
+        if qz == None: qz = self.current_pos.pose.orientation.z,
+        if qw == None: qw = self.current_pos.pose.orientation.w,
+
+        to_send = Waypoint()
+        # Set variables from built position
+        to_send.pose.position.x = x
+        to_send.pose.position.y = y
+        to_send.pose.position.z = z
+        to_send.pose.orientation.x = qx
+        to_send.pose.orientation.y = qy
+        to_send.pose.orientation.z = qz
+        to_send.pose.orientation.w = qw
+        # send goal to server to complete within timeout
+        move_result = self.send_goal(to_send, timeout)
+        return move_result
+
+    def relative_move(self, x=0, y=0, z=0, tx=0, ty=0, tz=0, timeout=10):
```

Comment:
a waypoint message is created on line 113, filled with data on the proceeding lines, and sent to the server on line 123.